File: swaggerdc/server/src/main/java/com/example/swaggerdc/services/CustomerService.java
package com.example.swaggerdc.services;

import com.example.swaggerdc.model.CustomerInformation;
import com.example.swaggerdc.model.User;
import com.example.swaggerdc.repository.CustomerInformationRepository;
import com.example.swaggerdc.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CustomerService {

  @Autowired
  private UserRepository userRepository;

  @Autowired
  private CustomerInformationRepository customerInformationRepository;

  public CustomerInformation findProfileByUsername(String username) {
    User user = userRepository.findFirstByUsername(username);
    System.out.print("FIND/Username - Encontrar cadastro do usuário");
    if (user != null) {
      return user.getCustomerInformation();
    }
    else {
      return null;
    }
  }

  public void updateCustomerInfo(int userId, CustomerInformation customerInfo) {
    User user = userRepository.findById(userId).orElse(null);
    System.out.println(
      String.format("UPDATE - Atualizar cadastro do usuário")
    );
    if (user != null) {
      CustomerInformation registeredInfo = user.getCustomerInformation();
      if (registeredInfo == null) {
        customerInfo.setId(userId);
        customerInformationRepository.save(customerInfo);
        user.setCustomerInformation(customerInfo);
        userRepository.save(user);
      } else {
        registeredInfo.setName(customerInfo.getName());
        registeredInfo.setEmail(customerInfo.getEmail());
        registeredInfo.setAddress(customerInfo.getAddress());
        customerInformationRepository.save(registeredInfo);
      }
    }
  }
}

File: adfund/src/main/kotlin/com/smdesenvolvimento/adfund/service/impl/CustomerService.kt
package com.smdesenvolvimento.adfund.service.impl

import com.smdesenvolvimento.adfund.model.Customer
import com.smdesenvolvimento.adfund.exception.BusinessException
import com.smdesenvolvimento.adfund.repository.CustomerRepository
import com.smdesenvolvimento.adfund.service.ICustomerService
import org.springframework.stereotype.Service

@Service
class CustomerService(
  private val customerRepository: CustomerRepository
): ICustomerService {
  override fun save(customer: Customer): Customer = this.customerRepository.save(customer)

  override fun findById(id: Long): Customer = this.customerRepository.findById(id)
    .orElseThrow{throw BusinessException("Id $id not found") }

  override fun delete(id: Long) {
    val customer: Customer = this.findById(id)
    this.customerRepository.delete(customer)
  }
}